ilastik/applets/objectExtraction/opObjectExtraction.py

```diff
@@ -708,19 +715,37 @@ def dictextend(a, b):
                     break
 
         if numpy.any(margin) > 0:
-            # starting from 0, we stripped 0th background object in global computation
-            for i in range(0, nobj):
-                logger.debug("processing object {}".format(i))
-                extent = self.compute_extent(i, image, mincoords, maxcoords, axes, margin)
-                rawbbox = self.compute_rawbbox(image, extent, axes)
-                # it's i+1 here, because the background has label 0
-                binary_bbox = numpy.where(labels[tuple(extent)] == i + 1, 1, 0).astype(bool)
-                for plugin_name, feature_dict in feature_names.items():
-                    if not has_local_features[plugin_name]:
-                        continue
-                    plugin = pluginManager.getPluginByName(plugin_name, "ObjectFeatures")
-                    feats = plugin.plugin_object.compute_local(rawbbox, binary_bbox, feature_dict, axes)
-                    local_features[plugin_name] = dictextend(local_features[plugin_name], feats)
+            bboxes = {}
+            for plugin_name, feature_dict in feature_names.items():
+                if not has_local_features[plugin_name]:
+                    continue
+
+                plugin = pluginManager.getPluginByName(plugin_name, "ObjectFeatures")
+                pool = RequestPool()
+                tmp_dicts = [None] * nobj
+
+                def _calc_single(i, raw_bbox, binary_bbox):
+                    feats = plugin.plugin_object.compute_local(raw_bbox, binary_bbox, feature_dict, axes)
+                    tmp_dicts[i] = feats
+
+                # starting from 0, we stripped 0th background object in global computation
+                for i in range(0, nobj):
+                    logger.debug("processing object {}".format(i))
+                    if i not in bboxes:
+                        extent = self.compute_extent(i, image, mincoords, maxcoords, axes, margin)
+                        raw_bbox = self.compute_rawbbox(image, extent, axes)
+                        # it's i+1 here, because the background has label 0
+                        binary_bbox = numpy.where(labels[tuple(extent)] == i + 1, 1, 0).astype(bool)
+                        bboxes[i] = (raw_bbox, binary_bbox)
+
+                    raw_bbox, binary_bbox = bboxes[i]
+                    pool.add(Request(partial(_calc_single, i, raw_bbox, binary_bbox)))
+
+                pool.wait()
+                pool.clean()
+                # merge the results
+                for i in range(0, nobj):
+                    local_features[plugin_name] = dictextend(local_features[plugin_name], tmp_dicts[i])
```

Comment:
```suggestion
                for feature_dict in tmp_dicts:
                    for feature_name, features in feature_dict.items():
                        local_features[plugin_name][feature_name].append(features)
```

...and there is no need for `dictextend`.

Reply:
Hm, yeah, you could inline it. I'm not sure it's better for readability here. Do you think so?

Reply:
It is used only in one place, the name doesn't explain what it actually does (at least to me), and the double loop has `dict1[key1][key2].append(value)` statement that shows how the main data structure is updated.